src/source_sampling.h

```diff
@@ -198,7 +206,14 @@ namespace pyne {
     int num_e_groups; ///< Number of groups in tag \a _src_tag_name
     int num_bias_groups; ///< Number of groups tag \a _bias_tag_name
     int max_num_cells; /// Max number of cells in voxels
+    // Equivalent cell number in photon source.
+    // For voxel R2S, p_src_num_cells = 1
+    // For sub-voxel R2S, p_src_num_cells = max_num_cells
+    // For unstructured R2S, p_src_num_cells = 1.
+    int p_src_num_cells;
+    int cell_list_size;
     BiasMode bias_mode; ///< Bias mode: ANALOG, UNIFORM, USER
+    int mode; ///< Sampler mode, currently support 0, 1, 2, 3, 4, 5
```

Comment:
Here it is (see my comment below on bias vs. sampling mode). This mode should be limited to only specifying `VOXEL` or `SUBVOXEL` the `bias_mode` attribute can be used to look up the bias when needed. This will simplify a lot of checks in the code I think.